### PR TITLE
Restore functionality to override the SDK using target.sdk-path

### PR DESCRIPTION
**Pull request by adrian-prantl**
<rdar://problem/62684243>

(sorry about the redundant cherry-picks)

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/test/API/lang/swift/xcode_sdk/TestSwiftXcodeSDK.py [hunk 1]
```diff
@@ -36,3 +33,32 @@ def test(self):
...
+    @swiftTest
+    @skipUnlessDarwin
+    @skipIfDarwinEmbedded
+    def test_override(self):
+        """Test that we can override the Xcode SDK using the target setting."""
+        self.build()
+        log = self.getBuildArtifact("types.log")
+        self.expect("log enable lldb types -f " + log)
+        ios_sdk = subprocess.check_output(
+            ['xcrun', '--show-sdk-path', '--sdk', 'iphonesimulator']
```
Comment (dcci) [inline lldb/test/API/lang/swift/xcode_sdk/TestSwiftXcodeSDK.py:46]: Does this work on a customer install?

Comment (adrian-prantl) [reply lldb/test/API/lang/swift/xcode_sdk/TestSwiftXcodeSDK.py:46]: I don't see why not, but we will know in ~1hour when the bot checked it.

Comment (adrian-prantl) [reply lldb/test/API/lang/swift/xcode_sdk/TestSwiftXcodeSDK.py:46]: (This is copied&pasted from another test)

Comment (dcci) [reply lldb/test/API/lang/swift/xcode_sdk/TestSwiftXcodeSDK.py:46]: Ship it then!

**Review by dcci — APPROVED** [review]

**Pull request merged**